src/to-be-visible.js

```diff
@@ -0,0 +1,39 @@
+import { matcherHint } from 'jest-matcher-utils';
+import { mergeAll } from 'ramda';
+
+import { checkReactElement, printElement } from './utils';
+
+function isStyleVisible(element) {
+  const style = element.props.style || {};
+  const { display = 'flex', opacity = 1 } = Array.isArray(style) ? mergeAll(style) : style;
+  return display !== 'none' && opacity !== 0;
+}
+
+function isAttributeVisible(element) {
+  return element.type !== 'Modal' || element.props.visible !== false;
```

Comment:
Directly checking modal type looks kinda hacky, maybe we could check some accessiblity props here: `accessibilityElementsHidden`, etc

Reply:
does it support bottom sheet from [gorhom](https://github.com/gorhom/react-native-bottom-sheet) ?

Reply:
Hard to say without checking, depends on how well that library supports accessibility props. I think we should first concentrate on supporting built-in components, e.g. built-in `Modal` from RN, and handle the external libraries afterwards on a case-by-case basis, as in some cases the proper place to support them will be here, but in other case it might be by submitting the PR to the library authors to improve their accessibility support.

Reply:
@fluiddot I wonder how `Modal` actually performs hidding of its children when `visibility=false`. Instead of relying on querying the element for type and `visible` prop, we could check the underlying code. Maybe it just does not render the children at all.

Reply:
> Instead of relying on querying the element for type and visible prop, we could check the underlying code. Maybe it just does not render the children at all.

@mdjastrzebski I checked when testing that `Modal` component doesn't actually render children, so this matcher will throw an exception when using it for children elements of a not visible modal.

Reply:
> Directly checking modal type looks kinda hacky, maybe we could check some accessiblity props here: accessibilityElementsHidden, etc

@mdjastrzebski good point. To be honest, I added the modal type check as a special case, since it's likely that we want to check the visibility of a modal using this matcher. In any case, I'm open to reconsider this condition in favor of a better approach or even not including it.

Regarding the accessibility props, I went ahead and updated the matcher to also check them in [this commit](https://github.com/testing-library/jest-native/pull/73/commits/33482121b8b15ff84aaf6a104c3197d18f70aa59), similar to the logic of [`isInaccessible` function](https://callstack.github.io/react-native-testing-library/docs/api#isinaccessible).